assetdb.go

```diff
@@ -0,0 +1,44 @@
+package assetdb
+
+import (
+	"github.com/owasp-amass/asset-db/backend"
+	"github.com/owasp-amass/asset-db/types"
+	oam "github.com/owasp-amass/open-asset-model"
+)
+
+type assetDB struct {
+	backend backend.Backend
+}
+
+func New(dbType backend.DBType, dsn string) *assetDB {
+	database := backend.New(dbType, dsn)
+	return &assetDB{
+		backend: database,
+	}
+}
+
+func (as *assetDB) Create(asset oam.Asset, srcAsset *types.StoredAsset, relationType *string) (types.StoredAsset, error) {
```

Comment:
Couple of things here:
1. please return a pointer to `StoredAsset` like `(*types.StoredAsset, error)`
2. Please re-order the parameters to be (source *StoredAsset, relation string, new oam.Asset)